co-authors-plus.php

```diff
@@ -2075,3 +2082,13 @@ function cap_get_coauthor_terms_for_post( $post_id ) {
 	global $coauthors_plus;
 	return $coauthors_plus->get_coauthor_terms_for_post( $post_id );
 }
+
+/**
+ * Register CoAuthor Blocks REST API Routes
+ */
+function cap_register_coauthors_blocks_rest_api_routes() : void {
```

Comment:
I made the API endpoints distinct from the rest of CAP and specific to the blocks based on input from Alec here:
https://github.com/Automattic/Co-Authors-Plus/issues/932#issuecomment-1601678110

Since I had just finished getting the user emails taken out of the `author` taxonomy endpoint in [this issue](https://github.com/Automattic/Co-Authors-Plus/issues/930), I was onboard to keep them "quite tied down".

As I explained in the questions section of my [initial comment about this pull request](https://github.com/Automattic/Co-Authors-Plus/pull/997#issue-1886352454), now that I've worked on it, I don't feel it needs to be private.

The API responses are view-only. In my opinion there's no sensitive information in them, and if we needed to add any later, it could be done with the `register_rest_field` action and permission controller on a per-value basis.

If we can make the view endpoints public, they could be used in de-coupled / headless CMS contexts like you described.

Reply:
@alecgeatches Any thoughts here?

Reply:
@GaryJones Where do we stand on this issue? Specifically I mean, can I update the endpoint that returns the coauthors for a post to make it public? I saw the mention in #1010 and I feel this is related. 

Making that public would make it possible to use a post id to find the coauthors.

It doesn't currently add anything to the REST API responses for coauthor-enabled post types, but that is possible too.

Reply:
@douglas-johnson @GaryJones I apologize for not following up on this thread! I have some time carved out early next week to do a full review, and hopefully get this merged in. I'll get back to you soon.

Reply:
@douglas-johnson @GaryJones I agree with both of you here. The `/coauthors` endpoint can have [the permissions guard](https://github.com/Automattic/Co-Authors-Plus/blob/5638d641eb7ae8ed469a25c3c857f7a427e40fcf/php/api/endpoints/class-coauthors-controller.php#L85) changed to `__return_true` here to remove permissions. We're only exposing [some basic properties](https://github.com/Automattic/Co-Authors-Plus/pull/997/files#diff-8fe0aa17984ddf192341f61e13907ab7510b2a50e89703c83571478001652be6R243) of the user and not leaking anything important like email addresses. Plus, as you mentioned above, this will help in decoupled/other API use-cases for pulling author information. Because this seems safe and useful, please feel free to move forward and open up the API response to unauthenticated users.

Reply:
@GaryJones I updated this in https://github.com/Automattic/Co-Authors-Plus/pull/997/commits/db08cc8608447e23c6f5eb6bc086e1cdb9545f71

I updated the `/coauthors/v1/coauthors?post_id=` endpoint to be public and utilized it in server-side code to get the array of co-authors so the rendered content matches the editor.

By making this public we make it possible to use in some headless applications, but the data isn't specifically linked to posts in the REST API; you kind of have to know its there. Can we make a separate issue to address the discoverability and ease of use? In my opinion there wasn't an obvious solution since the `coauthors` property for posts is already used for the `author` taxonomy.